### PR TITLE
chore(python): update Python to version 3.8.14

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Problem
We are running a version of Python (3.8.12) that has some security issues.

## Changes
This PR upgrades CI and our container images to [Python 3.8.14](https://www.python.org/downloads/release/python-3814/) released 2 days ago.

Note: as there isn't an image for Python 3.8.14 on Alpine 3.14, I've also updated the base image to 3.16 and as we are using a multistage build, I'm also upgrading the other base images and packages (yes, the change is way bigger than I thought). 

## How did you test this code?
CI is ✅

### CODE REVIEW

File: production.Dockerfile [hunk 1]
```diff
@@ -106,16 +106,15 @@ RUN apk --update --no-cache --virtual .geolite-deps add \
...
+    "gcc~=11.2" \
+    "cargo~=1.60" \
     "git~=2" \
     "make~=4.3" \
-    "libffi-dev~=3.3" \
+    "libffi-dev~=3.4" \
     "libxml2-dev~=2.9" \
     "libxslt-dev~=1.1" \
     "xmlsec-dev~=1.2" \
-    "postgresql-dev~=13" \
```
Comment (guidoiaquinti) [inline production.Dockerfile:118]: This package doesn't exist in Alpine 3.15 and 3.16. I'll get the error at build and see which other package we could use

Comment (guidoiaquinti) [reply production.Dockerfile:118]: It has been renamed to `postgresql13-dev`

File: production.Dockerfile [hunk 2]
```diff
@@ -66,14 +66,14 @@ WORKDIR /code
 # If you temporary need a package to build a Python or npm
 # dependency take a look at the sections below.
 RUN apk --update --no-cache add \
-    "libpq~=13" \
+    "libpq~=14" \
     "libxslt~=1.1" \
-    "nodejs-current~=16" \
```
Comment (guidoiaquinti) [inline production.Dockerfile:71]: There isn't a nodejs v16 package available so I had to bump it to 18

**Review by ellie — APPROVED** [review]
🚀

**Review by hazzadous — APPROVED** [review]

File: production.Dockerfile [hunk 3]
```diff
@@ -7,7 +7,7 @@
 #
 # Build the frontend artifacts
 #
-FROM node:16.15-alpine3.14 AS frontend
+FROM node:16.15-alpine3.16 AS frontend
```
Comment (hazzadous) [inline production.Dockerfile:10]: Should we also update to 18 here? and for plugin-server?

File: production.Dockerfile (hunk 2 above)
Comment (guidoiaquinti) [reply production.Dockerfile:71]: cc @yakkomajuri @tiina303 and @macobo for visibility

File: production.Dockerfile (hunk 3 above)
Comment (guidoiaquinti) [reply production.Dockerfile:10]: Yeah, I just wanted to have a 👍 from the ingestion team before doing so.

Comment (guidoiaquinti) [reply production.Dockerfile:10]: You are right. I'll make a PR

**Pull request merged**